aesara/tensor/math.py

```diff
@@ -1862,6 +1862,21 @@ def clip(x, min, max):
     # for grep: clamp, bound
 
 
+@scalar_elemwise
+def hyp2f1(a, b, c, z):
+    """gaussian hypergeometric function"""
+
+
+@scalar_elemwise
+def poch(z, m):
+    """pochhammer symbol (rising factorial) function"""
+
+
+@scalar_elemwise
+def factorial(n):
+    """factorial function"""
```

Comment:
The `scalar_elemwise` decorator will look for a `ScalarOp` instance with the name `poch`/`factorial`, and this is where the errors likely start.  Instead, if we remove these and define `poch` and `factorial` using the `Elemwise` version of `gamma` (i.e. the one created by `scalar_elemwise` in `aesara.tensor.math`) then everything should work.